src/pages/UniversalSwap/helpers.ts

```diff
@@ -94,49 +75,45 @@ export class UniversalSwapHandler {
     }
 
     const result = await transferIbcCustom(
-      this._toTokenSwap,
-      this._toTokenBridge ?? this._toToken,
+      this._toTokenInOrai,
+      this._toToken,
       simulateAmount,
       amounts,
       transferAddress
     );
     return result;
   }
 
-  async swap(
-    fromAmountToken: number,
-    simulateAmount: string,
-    oraiAddress: string,
-    userSlippage?: number
-  ): Promise<any> {
-    const messages = this.generateMsgsSwap(fromAmountToken, simulateAmount, userSlippage, oraiAddress);
+  async swap(fromAmountToken: number, simulateAmount: string, userSlippage?: number): Promise<any> {
+    const messages = this.generateMsgsSwap(fromAmountToken, simulateAmount, userSlippage);
     const result = await CosmJs.executeMultiple({
       prefix: ORAI,
-      walletAddr: oraiAddress,
+      walletAddr: this.sender,
       msgs: messages,
       gasAmount: { denom: ORAI, amount: '0' }
     });
     return result;
   }
 
-  // TODO
-  // swap orai from oraichain vs atom from cosmos
-  // => swap orai vs atom trên oraichain trước, sau đó bridge atom từ oraichain sang cosmos.
-  async swapAndTransfer(
-    fromAmountToken: number,
-    simulateAmount: string,
-    oraiAddress: string,
-    userSlippage?: number,
-    metamaskAddress?: string,
-    tronAddress?: string,
-    amounts?: AmountDetails
-  ): Promise<any> {
-    const swapResult = await this.swap(fromAmountToken, simulateAmount, oraiAddress, userSlippage);
+  /** Universal swap from Oraichain to cosmos-hub | osmosis | EVM networks.
+   * B1: find toToken in Oraichain (this._toTokenInOrai) to swap first
+   * B2: find toToken in in Osmosis | CosmosHub | Oraibridge for EVM network to bridge from Oraichain
+   */
+  async swapAndTransfer({
+    fromAmountToken,
+    simulateAmount,
+    userSlippage,
+    metamaskAddress,
+    tronAddress,
+    amounts
+  }: SwapData): Promise<any> {
+    // swap first
+    this._toTokenInOrai = oraichainTokens.find((t) => t.name === this._toToken.name);
+    const swapResult = await this.swap(fromAmountToken, simulateAmount, userSlippage);
```

Comment:
We need to join two transaction calls: swap & transfer into one single transaction. Otherwise, users will need to sign two times => not single-step anymore